### PR TITLE
feat (deployment): [jumpbox] remove public key from cloud init file

### PR DESCRIPTION
**Pull request by ferantivero**
- remove sensitive identifiers
- modify steps to ensure proper generation and assignment of the ssh public key

### CODE REVIEW

File: docs/deploy/07-aks-jumpbox-users.md [hunk 1]
```diff
@@ -15,26 +15,42 @@ Following these steps, you'll end up with an SSH public-key-based solution that
...
-      cat opsuser01.key.pub
-      ```
+1. Generate a key pair for this walk through
 
-      **Enter a passphrase when requested** (*do not leave empty*) and note where the public and private key file was saved. The *public* key file *contents* (`opsuser01.key.pub` in the example above) is what is added to the `ssh-authorized-keys` array in `jumpBoxCloudInit.yml`. You'll need the username, the private key file (`opsuser01.key`), and passphrase later in this walkthrough.
+   ```bash
+   ssh-keygen -t rsa -b 4096 -f opsuser01.key
+   ```
+
+   **Enter a passphrase when requested** (*do not leave empty*) and note where the public and private key file was saved. The *public* key file *contents* (`opsuser01.key.pub` in the example above) is what is added to the `ssh-authorized-keys` array in `jumpBoxCloudInit.yml`. You'll need the username, the private key file (`opsuser01.key`), and passphrase later in this walkthrough.
```
Comment (johndowns) [inline docs/deploy/07-aks-jumpbox-users.md:30]:
```suggestion
   **Enter a passphrase when requested** (*do not leave empty*) and note where the public and private key file was saved. The *public* key file *contents* (`opsuser01.key.pub` in the example above) will be used in the `jumpBoxCloudInit.yml` file. You'll need the username, the private key file (`opsuser01.key`), and passphrase later in this walkthrough.
```

File: docs/deploy/07-aks-jumpbox-users.md [hunk 2]
```diff
@@ -15,26 +15,42 @@ Following these steps, you'll end up with an SSH public-key-based solution that
...
    1. `lock_passwd:` - Leave at `True`. This disables password login, and as such the user can only connect via an SSH authorized key. Your jump box should enforce this as well on its SSH daemon. If you deployed using the image builder in the prior step, it does this enforcement there as well.
-   1. In `ssh-authorized-keys` replace the example public key for the user. This must be an RSA key of at least 2048 bits and **must be secured with a passphrase**. This key will be added to that user's `~/.ssh/authorized_keys` file on the jump box via the cloud-init bootstrap process. If you need to generate a key pair you can execute this command:
+   1. In `ssh-authorized-keys` replace the `<public-ssh-rsa-for-...>` placeholder with an actual public ssh public key for the user. This must be an RSA key of at least 2048 bits and **must be secured with a passphrase**. This key will be added to that user's `~/.ssh/authorized_keys` file on the jump box via the cloud-init bootstrap process.
 
-      ```bash
-      ssh-keygen -t rsa -b 4096 -f opsuser01.key
-      cat opsuser01.key.pub
-      ```
+1. Generate a key pair for this walk through
```
Comment (johndowns) [inline docs/deploy/07-aks-jumpbox-users.md:24]:
```suggestion
1. Generate an SSH key pair to use in this walkthrough.
```

File: docs/deploy/07-aks-jumpbox-users.md [hunk 3]
```diff
@@ -15,26 +15,42 @@ Following these steps, you'll end up with an SSH public-key-based solution that
...
+   ssh-keygen -t rsa -b 4096 -f opsuser01.key
+   ```
+
+   **Enter a passphrase when requested** (*do not leave empty*) and note where the public and private key file was saved. The *public* key file *contents* (`opsuser01.key.pub` in the example above) is what is added to the `ssh-authorized-keys` array in `jumpBoxCloudInit.yml`. You'll need the username, the private key file (`opsuser01.key`), and passphrase later in this walkthrough.
+
+   > On Windows, as an alternative to Bash in WSL, you can use a solution like PuTTYGen found in the [PuTTY installer](https://www.chiark.greenend.org.uk/~sgtatham/putty/latest.html).
+   >
+   > Azure also has an SSH Public Key resources type that allows you to [generate SSH keys](https://learn.microsoft.com/azure/virtual-machines/ssh-keys-portal) and keep public keys available as a managed resource.
+
+1. Manually add/remove/modify users following the information from the previous steps or simply execute the following command to overwrite the `jumpBoxCloudInit.yml` file with a valid user example
```
Comment (johndowns) [inline docs/deploy/07-aks-jumpbox-users.md:36]:
```suggestion
1. Run the following command to overwrite the `jumpBoxCloudInit.yml` file with a new user configuration that uses the SSH key you generated:
```

File: docs/deploy/07-aks-jumpbox-users.md [hunk 4]
```diff
@@ -15,26 +15,42 @@ Following these steps, you'll end up with an SSH public-key-based solution that
...
+   #cloud-config
+   users:
+     - default
+     - name: opsuser01
+       sudo: False
+       lock_passwd: True
+       ssh-authorized-keys:
+         - $(cat opsuser01.key.pub)
+   EOF
+   ```
```
Comment (johndowns) [inline docs/deploy/07-aks-jumpbox-users.md:49]: Alternatively, you can manually modify the existing `jumpBoxCloudInit.yml` file to update the users and add the key.

**Review by johndowns — APPROVED** [review]
Looks great! Just a few small suggestions to streamline some of the instructions slightly.

File: docs/deploy/07-aks-jumpbox-users.md (hunk 4 above)
Comment (ferantivero) [reply docs/deploy/07-aks-jumpbox-users.md:49]: definitely that is an option, yes!

Comment (johndowns) [reply docs/deploy/07-aks-jumpbox-users.md:49]: I meant that you could add wording like that here :)

**Review by johndowns — APPROVED** [review]
LGTM!

**Comment by ferantivero** [comment]
#sign-off

**Pull request merged**